Compile/src/Producers/PUJetIDWeightProducer.cc

```diff
@@ -0,0 +1,85 @@
+#include "Excalibur/Compile/interface/Producers/PUJetIDWeightProducer.h"
+#include "KappaTools/RootTools/interface/Matching.h"
+
+std::string PUJetIDWeightProducer::GetProducerId() const { return "PUJetIDWeightProducer"; }
+
+void PUJetIDWeightProducer::Init(ZJetSettings const& settings) {
+
+    PUJetID = settings.GetPUJetID();
+    PUJetIDSelectionMaxPt = 50.0;
+    inputIsData = settings.GetInputIsData();
+
+    if (inputIsData)
+    {
+        LOG(FATAL) << "PUJetID SFs can only be applied to MC!";
+    }
+
+    if (PUJetID != "none")
+    {
+        LOG(INFO) << "Apply PUJetID cut on jets pt<" << PUJetIDSelectionMaxPt << " GeV";
+        PUEffFilename = settings.GetPUJetIDEffFilename();
```

Comment:
The data type is not declared at this point. I guess this is not intentional.

Reply:
Was declared in the header, but not necessary to do so. Fixed it.